src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -939,7 +939,7 @@ def __repr__(self) -> str:
         return "<DaemonDescription>({type}.{id})".format(type=self.daemon_type,
                                                          id=self.daemon_id)
 
-    def to_json(self) -> dict:
+    def to_json(self, is_dict=False) -> dict:
```

Comment:
As said, I'm not super-excited about this approach, but the whole function is IMHO misleading (`to_json()` generating a dict with some JSON fields... 😒 ).

In any case, I'd change that a bit to make it more evident:
```suggestion
    def to_json(self, as_dict=False) -> dict:
```